cdap-api/src/main/java/co/cask/cdap/api/app/AbstractApplication.java

```diff
@@ -170,4 +173,16 @@ protected void scheduleWorkflow(Schedule schedule, String workflowName) {
   protected void scheduleWorkflow(Schedule schedule, String workflowName, Map<String, String> properties) {
     configurer.addSchedule(schedule, SchedulableProgramType.WORKFLOW, workflowName, properties);
   }
+
+
+  /**
+   * Schedule the specified {@link Workflow}
```

Comment:
This method actually doesn't schedule a workflow, right? This method is to return a `ScheduleConfigurer` and the workflow will only get scheduled when one of the `trigger` method is called, right? Maybe better calling it `configureSchedule`?? Also according to `SchedulableProgramType`, seems like not just workflow can be schedule? Although that class says `CUSTOM_ACTION` can be scheduled, which doesn't really make sense. Can you verify what that enum is for?

BTW, if we want to extend that to be able to schedule other programs, do we simply just add new methods?